code/datums/attack_style/_blocking.dm

```diff
@@ -0,0 +1,212 @@
+/**
+ * # Blocking
+ *
+ * Blocking incoming attacks, converting it to stamina damage.
+ */
+/datum/status_effect/blocking
+	id = "blocking"
+	alert_type = /atom/movable/screen/alert/status_effect/blocking
+	status_type = STATUS_EFFECT_REFRESH
+	tick_interval = 1 SECONDS
+	duration = -1
+
+	var/obj/item/blocking_with
+	var/obj/effect/blocking_effect/shield_overlay
+
+/datum/status_effect/blocking/nextmove_modifier()
+	// Next move CD is 2x as long while blocking.
+	// You are unable to attack while blocking but this handles stuff like
+	// interfacing with your backpack or other actions.
+	return 2
+
+/datum/status_effect/blocking/on_creation(mob/living/new_owner, obj/item/new_blocker)
+	. = ..()
+	if(!.)
+		return
+	if(!isnull(new_blocker))
+		set_blocking_item(new_blocker)
+
+	var/static/shield_offset_const = (0.8 * world.icon_size)
+	shield_overlay = new(new_owner)
+	shield_overlay.pixel_y += shield_offset_const // melbert todo: hides under mobs
+	owner.vis_contents += shield_overlay
+	update_shield()
```

Comment:
how hot is this code? is it worth trying to pool this object?

Reply:
I don't think it's *that* hot but it might be worth trying regardless, rather than re-instantiating it every time someone starts blocking

Reply:
disregard, not worth it. pooling has overhead